### DISCUSSION TITLE
Change deployment region of vercel serverless functions programmatically?

### DISCUSSION THREAD

**Discussion by aditodkar** [vercel/community#1233] [Help] [answered]
### Summary

Is there any way to programmatically change region of deployment of serverless functions instead of hardcoding it in vercel.json??

**Comment by leerob** [comment, marked as answer]
Can you share more about what you're trying to build?

**Reply by aditodkar** [reply]
Actually in my case I get more traffic from ap-south-1 during 9AM IST to 5PM IST and for 7PM IST to 6:30AM IST I get more traffic from us-east-1. So I wanted to know if we can change region via API or something else?

**Reply by amyegan** [reply]
When you [create a new deployment via API](https://vercel.com/docs/rest-api#endpoints/deployments/create-a-new-deployment) or [deploy via CLI](https://vercel.com/docs/cli/deploy#regions), you can specify the region for your site. But you would need to redeploy to move it to another region.

You can find more info about [Serverless Functions regions](https://vercel.com/docs/concepts/functions/serverless-functions/regions#) and [Edge Function regions](https://vercel.com/docs/concepts/functions/edge-functions#edge-function-regions) in the docs. A list of [valid region IDs](https://vercel.com/docs/concepts/edge-network/regions) is also there.
